app/components/Intercept_Components/InterceptTextBox.tsx

```diff
@@ -18,12 +20,22 @@ export const InterceptTextBox = props => {
         </a>
       </div>
       <div className="response">
-        <label htmlFor="">Response Text</label>
+        <label className="responseTextlabel">Response Text</label>
+        <span
+          className="fetch-responsetext"
+          onClick={() => {
+            props.fetchResponse(props.rowProps.checkbox);
+            textAreaValue = JSON.stringify(props.responseData[props.rowProps.checkbox.requestId]) || responseTextValue;
```

Comment:
this shouldn't be there. no need for `|| responseTextValue`. thats should be handled in store.

Reply:
`responseTextValue` holds the value that is propagated from `onChange event` of `textarea`. If there is nothing typed in the textarea, Then it holds a value of empty string. Here `textAreaValue` holds the value that is propagated from the store as `props` when the `X(fetch response)` button is clicked. We could handle this in a `if else` conditional, but not in a store I guess. Any other approach you can think of. Please refer : https://github.com/code-mancers/interceptor/blob/18a38363d3f4f4700cc6c6e40fa25f875f528d2d/app/components/Intercept_Components/InterceptTextBox.tsx

Reply:
okey I understand the `|| responseTextvalue` part. but still this statement should not be inside the `onClick.` This statement should be in the `render` method.